### DISCUSSION TITLE
Modify strategy with middleware

### DISCUSSION THREAD

**Discussion by AshotN** [floating-ui/floating-ui#2343] [Q&A]
I have a middleware like this

```ts
const mobilePosition: Middleware = {
    name: "middleware",
    async fn(state) {
      const overflow = await detectOverflow(state);
      if (overflow.left === 0 && overflow.right === 0) {
        return { x: 0, y: 0, strategy: "fixed" };
      }
      return {};
    },
  };
```

I want to be able to set `strategy` from the return value. My use case is that on desktop the UI is not full-screen and does not need to be fixed, while on mobile it takes the whole screen and should be fixed.

Is there anyway to achieve this currently?

**Comment by atomiks** [comment]
This can lead to loops in certain cases, so it's better to do it externally instead. You can use a media query in JS to set it to `fixed` at a certain screen size - with the limitation that it's not fully dynamic.

**Comment by AshotN** [comment]
Currently am using media-querys in CSS to over-ride the position element. But it's kind of ugly, if there is no better solution then that's what I'll keep.
